File: Rearrange Array Elements by Sign.go
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> arr;
        arr.reserve(n);
        int i = 0;
        int j = 0;
        bool pos = true;
        while (nums[i] < 0) {
            i++;
        }
        while (nums[j] > 0) {
            j++;
        }

        while (i < n || j < n) {
            if (pos) {
                arr.push_back(nums[i++]);
                while (i < n && nums[i] < 0) {
                    i++;
                }
            } else {
                arr.push_back(nums[j++]);
                while (j < n && nums[j] > 0) {
                    j++;
                }
            }
            pos = !pos;
        }
        return arr;
    }
};

File: Rearrange Array Elements by Sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        arr = []
        pos = True
        n = len(nums)
        i, j = 0, 0
        while nums[i] < 0:
            i += 1

        while nums[j] > 0:
            j += 1

        while i < n and j < n:
            if pos:
                arr.append(nums[i])
                i += 1
                while i < n and nums[i] < 0:
                    i += 1
            else:
                arr.append(nums[j])
                j += 1
                while j < n and nums[j] > 0:
                    j += 1
            pos = not pos

        if j < n:
            arr.append(nums[j])
        else:
            arr.append(nums[i])

        return arr
